gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/Operator.java

```diff
@@ -113,6 +133,11 @@ public Object apply(final Object a, final Object b) {
         public Object apply(final Object a, final Object b) {
             return b;
         }
+
+        @Override
+        public boolean isCommutative() {
```

Comment:
I believe assignment should be non-commutative.

Reply:
isCommutative is a confusing name, `assign` may be also the one but it is already used in GraphComputer's code internally. I just want to exclude `div` and `minus` at the moment, let's create a static list to disable execution for them.